lib/chef/pwsh.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Author:: Matt Wrock (<mwrock@chef.io>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class Chef
+  class Pwsh < Chef::PowerShell
+
+    @@wrapper_dll ||= Dir.glob("#{RbConfig::CONFIG["bindir"]}/**/Chef.PowerShell.Wrapper.Core.dll").last
```

Comment:
Rather than use a class variable for this and overwrite the initializer, I think I'd rather see it as a class method that's called in the superclass's `exec` or make it an argument that gets passed to that function. Class variables in general have kind of strange behavior and I try to avoid them, and since this is doing "work" it could be made lazily memoized to avoid having to pay that cost just for loading the file.

Reply:
yeah this felt totally weird. sadly all of my ruby knowledge around scoping and the proper way to do class initialization has atrophied. will make this better.

Reply:
ok I moved it from a class variable to a class method. I still had to override the instance variable because I don't think you can override a class method in a subclass??